### PR TITLE
support more type in postgres-extended

### PR DESCRIPTION
**Pull request by ZENOTME**
This PR does two things:
1. support INTERVAL && TIMESTAMPTZ type in postgres-extended 
2. refactor run() in postgres-extended: use a macro to generate process

### CODE REVIEW

File: sqllogictest-bin/src/engines/postgres_extended.rs [hunk 1]
```diff
@@ -113,115 +217,21 @@ impl sqllogictest::AsyncDB for PostgresExtended {
...
-                            array_process!(row, output, idx, f64);
-                        }
-                        Type::NUMERIC_ARRAY => {
-                            array_process!(row, output, idx, Decimal);
-                        }
-                        _ => {
-                            todo!("Don't support {} type now.", column.type_().name())
-                        }
-                    }
+                    generate_process!(column,row, output,idx,
```
Comment (xxchan) [inline sqllogictest-bin/src/engines/postgres_extended.rs:220]: Why to use macro here? How is it simplier than before?

Comment (ZENOTME) [reply sqllogictest-bin/src/engines/postgres_extended.rs:220]: reduce duplication like
```
Type::NUMERIC => {
    single_process!(row, output, idx, Decimal);
}
Type::TIMESTAMP => {
    single_process!(row, output, idx, NaiveDateTime);
}
Type::DATE => {
    single_process!(row, output, idx, NaiveDate);
}
...

```

Comment (xxchan) [reply sqllogictest-bin/src/engines/postgres_extended.rs:220]: IMO such level of duplication is fine. Macro seems an overkill and hurts readability.

**Review by xxchan — APPROVED** [review]

**Review by xxchan — COMMENTED** [review]
BTW, why would you want to format the time results instead of keeping it as-is? What's the motivation? 👀

**Comment by ZENOTME** [comment]
> BTW, why would you want to format the time results instead of keeping it as-is? What's the motivation? 👀

Because for the time type(such as INTERVAL), there is more than one string format representing the same value. 
Such as '31 days' and '1 mon', they have the same semantics.
So the easiest way to compare these is to declare a fixed format.

More detail: #93

**Comment by xxchan** [comment]
> there is more than one string format representing the same value

But *maybe* some users do want to depend on such behavior, and the db implementation want to respect the behavior? (😅

**Comment by xxchan** [comment]
BTW this will make `postgres` and `postgres-extended` not compatible. i.e., some test cases will pass one but fail the another. Is this what you expect?

**Comment by ZENOTME** [comment]
> BTW this will make postgres and postgres-extended not compatible. i.e., some test cases will pass one but fail the another. Is this what you expect?

In risingwave, Postgres and Postgres-extended have different e2e tests now. So I think it's acceptable to make it not compatible now. After introducing the custom comparator, we can make it compatible I think. 🤔

**Comment by xxchan** [comment]
> After introducing the custom comparator, we can make it compatible I think.

Actually it's still not 100% "compatible". Specifically if it can pass `postgres-extended`, it can fail `postgres` https://github.com/risinglightdb/sqllogictest-rs/issues/93#issuecomment-1288110789 🤔

Maybe we need to completely switch to `postgres-extended`?

**Comment by ZENOTME** [comment]
> Maybe we need to completely switch to postgres-extended?

We can't completely switch to postgres-extended because there are some type can't support in postgres-extended such as multi-dimensional array and struct.

**Review by skyzh — APPROVED** [review]
Originally I think of rewriting all of SQL queries that output date time column, so that they won't directly appear in sqllogictest. e.g, `to_char(interval '15h 2m 12s', 'HH24:MI:SS')`. Anyway, this approach also LGTM.

**Comment by skyzh** [comment]
Ping me when you're ready to merge and release a new version.

**Comment by ZENOTME** [comment]
> Originally I think of rewriting all of SQL queries that output date time column, so that they won't directly appear in sqllogictest. e.g, `to_char(interval '15h 2m 12s', 'HH24:MI:SS')`. Anyway, this approach also LGTM.

I'm aware of actually we can use `(...)::varchar eg: (interval '15h 2m 12s')::varchar`  to convert any type into varchar type so that we can compare it with string.

**Comment by ZENOTME** [comment]
> I'm aware of actually we can use (...)::varchar eg: (interval '15h 2m 12s')::varchar to convert any type into varchar type so that we can compare it with string.

I use this way to convert the` interval timestamptz `type to string. So now for these two types, Postgres and Postgres-extended are compatible. And I also do other additional work:
- support array type of the supported single type
- add a test file 

@skyzh @xxchan

**Comment by skyzh** [comment]
LGTM, merging now.

**Pull request merged**